plugins/module_utils/ec2.py

```diff
@@ -648,6 +648,58 @@ def create_dhcp_options(
     return client.create_dhcp_options(**params)["DhcpOptions"]
 
 
+# EC2 vpn Gateways
+class EC2VpnGatewaysErrorHandler(AWSErrorHandler):
+    _CUSTOM_EXCEPTION = AnsibleEC2Error
+
+    @classmethod
+    def _is_missing(cls):
+        return is_boto3_error_code(["InvalidVpnGatewayID.NotFound", "InvalidVpnGatewayState"])
+
+
+@EC2VpnGatewaysErrorHandler.list_error_handler("describe vpn gateways", [])
+@AWSRetry.jittered_backoff()
+def describe_vpn_gateways(
+    client, **params: Dict[str, Union[List[str], int, List[Dict[str, Union[str, List[str]]]]]]
+) -> List[Dict[str, Any]]:
+    return client.describe_vpn_gateways(**params)["VpnGateways"]
+
+
+@EC2VpnGatewaysErrorHandler.common_error_handler("create vpn gateway")
+@AWSRetry.jittered_backoff()
```

Comment:
```suggestion
@AWSRetry.jittered_backoff(catch_extra_error_codes=['VpnGatewayLimitExceeded'])
```
This is useful to replace the class `VGWRetry` behavior used in the `ec2_vpc_vgw` module.